buf/registry/plugin/v1beta1/plugin_service.proto

```diff
@@ -0,0 +1,229 @@
+    repeated PluginLanguage languages = 3 [
+      (buf.validate.field).repeated.unique = true,
+      (buf.validate.field).repeated.items.enum.defined_only = true,
+      (buf.validate.field).repeated.items.enum.not_in = 0
+    ];
+    // The registry types to list Plugins for.
+    //
+    // If empty, all Plugins are listed regardless of registry type.
+    repeated PluginRegistryType registry_types = 4 [
+      (buf.validate.field).repeated.unique = true,
+      (buf.validate.field).repeated.items.enum.defined_only = true,
+      (buf.validate.field).repeated.items.enum.not_in = 0
+    ];
+  }
+  // The maximum number of items to return.
+  //
+  // The default value is 10.
+  uint32 page_size = 1 [(buf.validate.field).uint32.lte = 250];
+  // The page to start from.
+  //
+  // If empty, the first page is returned.
+  string page_token = 2 [(buf.validate.field).string.max_len = 4096];
+  // An optional filter to filter the list of Plugins. If unset, all Plugins for all organizations
+  // are listed, but this functionality is limited to Users with the necessary permissions.
+  Filter filter = 3;
+  // The order to return the Plugins.
+  //
+  // If not specified, defaults to ORDER_CREATE_TIME_DESC.
+  Order order = 4 [(buf.validate.field).enum.defined_only = true];
+}
+
+message ListPluginsResponse {
+  // The next page token.
+  //
+  // If empty, there are no more pages.
+  string next_page_token = 1 [(buf.validate.field).string.max_len = 4096];
+  // The listed Plugins.
+  repeated Plugin plugins = 2;
+}
+
+message CreatePluginsRequest {
+  // An individual request to create a Plugin.
+  message Value {
+    // The Organization to create the Plugin under. The OwnerRef must be a reference to an
+    // Organization and cannot be a User reference.
+    buf.registry.owner.v1.OwnerRef owner_ref = 1 [(buf.validate.field).required = true];
+    // The name of the Plugin.
+    string name = 2 [(buf.validate.field).string = {
+      min_len: 2
+      max_len: 100
+    }];
+    // The plugin's visibility.
+    PluginVisibility visibility = 3 [
+      (buf.validate.field).required = true,
+      (buf.validate.field).enum.defined_only = true
+    ];
+    // The configurable description of the Plugin.
+    string description = 4 [(buf.validate.field).string.max_len = 350];
+    // The configurable source URL of the Plugin.
+    //
+    // TODO(mf): do we want this at the Plugin or the PluginCommit.
+    //
+    // I'm inclined to leave this on the Plugin. I think the most common use case is a plugin
+    // repository moving from github.com/foo/bar to github.com/new/repo and that event should apply
+    // to all commits since the project as a whole is moving, and not a subset of commits.
+    //
+    // For example, connect used to be under bufbuild but moved to a new organization, when looking
+    // at the Go docs there is a single source of truth for the project despite having a different
+    // home.
+    string source_url = 5 [
```

Comment:
IMO the `source_url` should be a property of the plugin entity and not the commit, which means a user can update this via the BSR and have it applied to _all_ commits.

> I think the most common use case is a plugin repository moving from github.com/foo/bar to
> github.com/new/repo and that **event should apply to all commits since the project as a whole is
> moving, and not a subset of commits.**
> 
> For example, connect used to be under bufbuild but moved to a new organization, when looking at
> the Go docs there is a single source of truth for the project despite having a different home.

There's a `url` in the PluginInfo, but we should ignore this value in favor of whatever is set on the BSR.

https://buf.build/bufbuild/bufplugin/docs/main:buf.plugin.info.v1#buf.plugin.info.v1.PluginInfo